lib/vfs/src/special_file.rs

```diff
@@ -0,0 +1,98 @@
+//! Used for /dev/stdin, /dev/stdout, dev/stderr - returns a
+//! static file descriptor (0, 1, 2)
+
+use std::io::{self, *};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use crate::VirtualFile;
+use tokio::io::{AsyncRead, AsyncSeek, AsyncWrite};
+use wasmer_wasi_types::wasi::Fd;
+
+/// A "special" file is a file that is locked
+/// to one file descriptor (i.e. stdout => 0, stdin => 1), etc.
+#[derive(Debug)]
+pub struct SpecialFile {
```

Comment:
I believe this have a specific naming in unix? (although forgot which is the correct one!)

Reply:
It's often called special file, sometimes device file.

Reply:
But the system here just pins to one specific `Fd`, so it's a bit more general than "device file".

I think the name is fine.

Reply:
Ok, after thinking more about this, I think we should name this `Device` and then stdin/stderr/stdout are just special kind of devices?

Reply:
As mentioned, I don't think `DeviceFile` makes sense here, since it doesn't need to be a device at all.